src/trap.S

```diff
@@ -2,11 +2,27 @@
 // Licensed under the Apache License, Version 2.0, see LICENSE for details.
 // SPDX-License-Identifier: Apache-2.0
 
-/// Trap handler entry point.
+.attribute arch, "rv64gcv"
 .section .text.init
+
+// Early trap handler. Doesn't check for stack overflow as hypervisor mappings are not enabled yet.
+.global _trap_init_entry
+.balign 4
+_trap_init_entry:
+        /* Skip the buffer overflow check at init time. */
+        j .L_trap_handler
```

Comment:
nit: inconsistent indentation between this block and everything below.

Reply:
Ah yes, sorry. Somehow the indentation in my emacs configuration is really incompatible with the way asm is formatted here, so it's all manual. Will fix.